### DISCUSSION TITLE
Creating FormData server-side

### DISCUSSION THREAD

**Discussion by latchk3y** [pocketbase/pocketbase#4544] [Q&A] [answered]
obviously, client-side, it's very easy to simply call new FormData() and just be done with it.

however, in order to do something like use data in a cronJob to upload to a SEPERATE api (mastodon, let's say), you'd be required to upload the file using formData as well. 

so short version of the question, what is the pocketbase equivalent of new FormData(), so you could make such API post requests?

**Comment by ganigeorgiev** [comment]
There is no support at the moment for sending `multipart/form-data` requests with the JSVM `$http.send` at the moment (_or at least not directly since it will require you to manually construct the raw body including the fields boundary/delimiter_).

I'll consider adding support for it after the refactoring.

**Reply by ganigeorgiev** [reply]
I forgot to mention that one possible workaround could be to send the request with `curl`, `wget` or other common OS utility using the [`$os.cmd()`](https://pocketbase.io/jsvm/functions/_os.cmd.html) helper.

_In any case, since the refactoring could take a while and this was asked several times before, I'll try sometime later this week to see how feasible would be to create a `FormData` equivalent JSVM API integrated with the existing [`$http`](https://pocketbase.io/jsvm/modules/_http.html) and [`$filesystem`](https://pocketbase.io/jsvm/modules/_filesystem.html) abstractions._

**Reply by ganigeorgiev** [reply, marked as answer]
_Just for info in case someone stumble here - `FormData` and support for sending `multipart/form-data` requests with `$http.send` was added in [v0.22.4 release](https://github.com/pocketbase/pocketbase/releases/tag/v0.22.4). For example you can refer to the [updated docs](https://pocketbase.io/docs/js-sending-http-requests/#multipartform-data-requests)._

**Reply by latchk3y** [reply]
well, that was fast.
